lib/Doctrine/Common/Collections/Collection.php

```diff
@@ -197,6 +243,10 @@ public function forAll(Closure $p);
      * a new collection with the elements returned by the function.
      *
      * @return Collection
+     *
+     * @template U
+     * @psalm-param Closure(T):U $func
+     * @psalm-return Collection<array-key, U>
```

Comment:
```suggestion
     * @psalm-return Collection<TKey, U>
```
Map is expected to keep keys.